specs/BrowserProcessExited.md

```diff
@@ -0,0 +1,312 @@
+  ///
+  /// All `CoreWebView2Environment` objects across different app processes that use
+  /// the same browser process receive this event when the browser process exits.
+  /// If the browser process (and therefore the user data folder) in use by the app
+  /// process (through the `CoreWebView2Environment` options used) is shared with
+  /// other processes, these processes need to coordinate to handle the potential race
+  /// condition on the use of the resources. E.g., if one app process tries to clear
+  /// the user data folder, while other tries to recreate its WebViews on crash.
+  ///
+  /// Note this is an event from the `ICoreWebView2Environment3` interface, not the
+  /// `ICoreWebView2`. The difference between this `BrowserProcessExited` event and
+  /// the `CoreWebView2`'s `ProcessFailed` event is that `BrowserProcessExited` is
+  /// raised for any (expected and unexpected) **browser process** exits, while
+  /// `ProcessFailed` is raised only for **unexpected** browser process exits, or for
+  /// **render process** exits/unresponsiveness. To learn more about the WebView2
+  /// Process Model, go to [Process model](https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/process-model).
+  ///
+  /// In the case the browser process crashes, both `BrowserProcessExited` and
+  /// `ProcessFailed` events are raised, but the order is not guaranteed.
+  HRESULT add_BrowserProcessExited(
+		  [in] ICoreWebView2BrowserProcessExitedEventHandler* eventHandler,
+		  [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_BrowserProcessExited`.
+  HRESULT remove_BrowserProcessExited([in] EventRegistrationToken token);
+}
+
+/// Receives `BrowserProcessExited` events.
+interface ICoreWebView2BrowserProcessExitedEventHandler : IUnknown
+{
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+		  [in] ICoreWebView2Environment* sender,
+		  [in] ICoreWebView2BrowserProcessExitedEventArgs* args);
+}
+
+/// Event args for the `BrowserProcessExited` event.
+interface ICoreWebView2BrowserProcessExitedEventArgs : IUnknown
+{
+  /// The kind of browser process exit that has occurred.
+  [propget] HRESULT BrowserProcessExitKind(
+      [out, retval] COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND* browserProcessExitKind);
+}
+```
+
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    // ...
+
+    /// Specifies the browser process exit kind used in
+    /// `CoreWebView2BrowserProcessExitedEventArgs`.
+    enum CoreWebView2BrowserProcessExitKind
+    {
+        /// Indicates that the browser process ended normally.
+        NormalExit,
+        /// Indicates that the browser process ended unexpectedly.
+        /// A `CoreWebView2.ProcessFailed` event will also be raised to
+        /// listening WebViews from the `CoreWebView2Environment` associated to
+        /// the failed process.
+        FailedExit
```

Comment:
I can't comment in C++ signatures, but for .NET the names appears to be unnecessary verbose, e.g. would the following be less clearer to a developer?
```cs
    enum BrowserProcessExitKind
    {
        /// Indicates that the browser process ended normally.
        Normal,
        /// Indicates that the browser process ended unexpectedly.
        /// A `CoreWebView2.ProcessFailed` event will also be raised to
        /// listening WebViews from the `CoreWebView2Environment` associated to
        /// the failed process.
        Failed
```

Reply:
Thanks for your feedback!
These enum values and names are still under consideration. We are definitively taking your input.

There might still be more changes to this API proposal before it's ready for a broader review audience. We will post the API Review in our [WebView2Announcements](https://github.com/MicrosoftEdge/WebView2Announcements/issues?q=is%3Aopen+is%3Aissue+label%3A%22API+Review%22) repo when it's ready. Feel free to leave your feedback, just note this is an early draft of the API.

Reply:
Its at least a requirement by API review naming to do it this way for WinRT and we want our .NET and WinRT APIs to match when possible. Its also consistent with our other enums.

Reply:
Oh I misunderstood the original issue. The change to Normal/Failed looks good